web/src/features/charts/graphUtils.ts

```diff
@@ -112,6 +113,31 @@ export const getNextDatetime = (datetimes: Date[], currentDate: Date) => {
   return datetimes[index + 1];
 };
 
+export function determineUnit(
+  displayByEmissions: boolean,
+  currentZoneDetail: ZoneDetail,
+  mixMode: Mode,
+  timeAverage: TimeAverages
+) {
+  if (displayByEmissions) {
+    return getUnit(
+      formatCo2(getTotalEmissionsAvailable(currentZoneDetail, mixMode)) +
+        ' ' +
+        t('ofCO2eq')
+    );
+  }
+
+  return timeAverage === TimeAverages.HOURLY
+    ? getUnit(formatPower(getTotalElectricityAvailable(currentZoneDetail, mixMode)))
+    : getUnit(formatEnergy(getTotalElectricityAvailable(currentZoneDetail, mixMode)));
+}
+
+function getUnit(valueAndUnit: any) {
+  const regex = /\s+(.+)/;
+  const match = valueAndUnit.match(regex);
+  return match[1];
+}
+
```

Comment:
I couldn't figure out a good way to determine the unit of the BarBreakDownChart, so it is a bit hacky... Suggestions are very welcome:)

Reply:
I think it's fine to do it this way.

The alternative would be to refactor the logic in `formatPower` and `formatEnergy` functions so we have a separate function for getting the unit which we could then reuse here (see pseudo-code example below), but I don't think that's necessarily worth it for this :)


```js
function calculatePowerUnit(value) {
  // Move logic from formatPower here
}
function formatPower(value) {
  const formattedValue = ...
  const unit = calculatePowerUnit(formattedValue)
  return `${formattedValue} ${unit}`
}

Reply:
This might be a better approach, but the `formatPower` uses `d3-format` which adds the M, G, depending on the number, before W so it kind of does both things at the same time. So I couldn't really figure out a way to separate it into two different functions.

Reply:
Let's keep it like this for now and maybe add a task for it either the next cycle if you think it's important enough or just as a slack week item.